### PR TITLE
RHDG terminology

### PR DESCRIPTION
**Pull request by dvagnero**
add data grid terms to the glossary
https://issues.redhat.com/browse/JDG-4940

### CODE REVIEW

**Comment by dvagnero** [comment]
@oraNod could you please review?

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/c.adoc
Comment (oraNod) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/c.adoc:359]:
```suggestion
*Description*: A configuration that allows Data Grid clusters to form a global view and back up data across geographically disperse locations. Multiple clusters running in different data centers replicate data between each other to ensure business continuity in the event of an outage and to present a single, unified caching service to applications.
```

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/c.adoc
Comment (oraNod) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/c.adoc:363]:
```suggestion
*Incorrect forms*: xsite
```

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (oraNod) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:49]:
```suggestion
*Description*: "In-memory" systems store data in a computer's main memory, random access memory (RAM). Clusters share memory resources, which reduces waste and boosts application performance by providing access to data in the same memory space where code executes.
```

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc
Comment (oraNod) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc:37]:
```suggestion
*Description*: A type of storage that persists data even after the computer is turned off or after the process that created the data ended, for example hard disk, tape, or flash memory.
```

Comment (oraNod) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc:37]: s/them/the data because it wasn't clear what the pronoun was referencing. Also I'm a bit of a stickler for the Oxford comma.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/v.adoc
Comment (oraNod) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/v.adoc:208]:
```suggestion
*Description*: Temporary storage, for example random access memory (RAM), that requires power to maintain the stored information. The stored data is lost when the device is turned off or restarts.
```

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (oraNod) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:8]:
```suggestion
*Description*: Interface for creating caches and managing cache lifecycles. Always spell as two words with capital letters when referring to the abstract notion of a Cache Manager. When referring to specific interfaces, use `CacheManager`, `EmbeddedCacheManager`, or `RemoteCacheManager`.
```

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (oraNod) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:52]:
```suggestion
*Description*: Hot Rod is a binary TCP client-server protocol used in Red Hat Data Grid. Java, C#, C++, NodeJS clients, as well as clients written in other programming languages, and clients can access data that resides in remote caches on Data Grid Server clusters via the Hot Rod endpoint. Write as two words and capitalize the first letter of each word.
```

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (oraNod) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:56]:
```suggestion
*Incorrect forms*: hot rod, HotRod, hotrod
```

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (oraNod) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:63]:
```suggestion
*Description*: Infinispan is the open-source, community project on which Red Hat Data Grid is built.
```

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (oraNod) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:78]:
```suggestion
*Incorrect forms*: Red Hat JBoss Data Grid, JDG
```

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (oraNod) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:96]:
```suggestion
*Description*: Data Grid clusters can provide external cache containers that store application specific data. These external caches store HTTP sessions and other data to make applications stateless and achieve elastic scalability as well as high-availability.
```

**Review by oraNod — CHANGES_REQUESTED** [review]
Great work @dvagnero Thanks so much for adding these RHDG terms. I've made a few suggestions. Please let me know if you've any questions or follow ups. Cheers.

**Review by oraNod — DISMISSED** [review]
LGTM

**Comment by oraNod** [comment]
@inoxx03 or @bergerhoffer Any review comments or OK to merge?

**Comment by dvagnero** [comment]
Thanks for your review and adding the appropriate labels and reviewers @oraNod :)

**Comment by bergerhoffer** [comment]
@oraNod thanks for reviewing! We require 3 approvals from council members before merging. I'll go ahead and review now!

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:19]: Is there any guidance for using this only after introducing the full name first that we should include here?

An example of an entry that does this now: https://redhat-documentation.github.io/supplementary-style-guide/#red-hat-cloudforms

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:52]: Is "NodeJS" correct? I always thought it was written as "Node.js"

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:74]: Same comment as for the short name. Should there be any guidance that the short is only to be used after the full name was introduced first? Or can people use either whenever they want?

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:96]: * application specific -> application-specific
* high-availability -> high availability

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/v.adoc
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/v.adoc:207]: Can "Volatile" be lowered to "volatile" (since I assume it doesn't have to start with a capital V if it's in the middle of a sentence)?

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc:43]: Volatile -> volatile

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/c.adoc
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/c.adoc:358]: Cross -> cross

**Review by bergerhoffer — COMMENTED** [review]
Thanks a lot for submitting this update @dvagnero, it looks great! Just a few comments when you get a chance to take a look. Let me know if you have any questions on any of them.

**Review by inoxx03 — DISMISSED** [review]
Thank you for adding the RHDG terms @dvagnero

**Comment by dvagnero** [comment]
Great points. Thanks for your review @bergerhoffer. Hopefully I didn't miss anything

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc
Comment (ndeevy) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc:37]: I think persist is an intransitive verb, in which case the storage can't _persist_ the data. It could preserve it or retain it though because those are transitive. Also in another PR we changed the glossary entries to complete sentences, which would make sense throughout this one too. So here maybe something like this would work:

`Persistent storage is type of storage that preserves data even after the computer is turned off or after the process that created the data ends. A hard disk, tape, or flash memory are examples of persistent storage.`

This suggestion also makes the last clause a separate sentence because it read as if those were examples of data ending, not persistent storage itself.

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (ndeevy) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:52]:
```suggestion
*Description*: Data Grid Server is a standalone server that exposes any number of caches to clients over a variety of protocols, including Hot Rod, Memcached and REST. Always capitalize when referring to a Data Grid Server instance.
```

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (ndeevy) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:63]: I'm getting a smidge tripped up parsing this sentence. It seems like there's something missing. Could you try to simplify it a bit? (it might be just my brain today though!)

`Java, C#, C++, Node.js clients, as well as clients written in other programming languages, and clients can access data that resides in remote caches on Data Grid Server clusters via the Hot Rod endpoint.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/v.adoc
Comment (ndeevy) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/v.adoc:208]: To make this a complete sentence you could say something like:

`Volatile storage is a temporary type of storage, for example, random access memory (RAM) that requires power to maintain the stored information. The stored data is lost when the device is turned off or restarts.`

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc
Comment (ndeevy) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-data-grid.adoc:8]: [GLOBAL] comment: I suggest to go through and change all the entries into complete sentences.

**Comment by ndeevy** [comment]
Review complete @dvagnero!
Looking good overall but I added a few small comments. Reply if you have any Qs about them. ;-]
Thanks

**Review by oraNod — APPROVED** [review]
Thanks @dvagnero

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/v.adoc
Comment (ndeevy) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/v.adoc:208]:
```suggestion
*Description*: Volatile storage is temporary storage, for example, random access memory (RAM) that requires power to maintain the stored information. The stored data is lost when the device is turned off or restarts.
```
Great improvement!  just need to remove this leftover 'a'

**Comment by ndeevy** [comment]
Looks excellent now, just one stray typo but otherwise LGTM @dvagnero :-]

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/v.adoc
Comment (dvagnero) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/v.adoc]: Thanks @ndeevy . Now it should be ok

**Review by ndeevy — APPROVED** [review]

**Review by inoxx03 — APPROVED** [review]

**Review by sterobin — APPROVED** [review]

**Comment by inoxx03** [comment]
We have 3 ACKs from the Style Council, I am merging this PR! Thanks, all:)

**Pull request merged**
